pkg/project/util/util.go

```diff
@@ -27,6 +29,24 @@ func ConvertNamespace(namespace *kapi.Namespace) *projectapi.Project {
 	}
 }
 
+// ConvertNamespace transforms a Namespace into a Project
+func ConvertNamespaceExternalOrDie(namespace *corev1.Namespace) *projectapi.Project {
+	internalNs := &kapi.Namespace{}
+	if err := kapiv1.Convert_v1_Namespace_To_core_Namespace(namespace, internalNs, nil); err != nil {
```

Comment:
don't convert to an internal ns and eliminate the possibility of an error.

Reply:
That would mean I'd have to return a `projectv1.Project` object. I am assuming that this would be okay, since the consumer of this function is a `watch.Event`?

Reply:
hm, this would mean that we would be returning v1 projects for some events, and non-versioned projects for others

Reply:
```go
internalFinalizers := rightTpe here
for _, externalFinalizer := externalNS.spec.Finalizers{
    internalFinalizers = append(internalFinalizers, internalFinalizer(string(externalFinalizer)))
}
return &projectapi.Project{
ObjectMeta: namespace.ObjectMeta,
		Spec: projectapi.ProjectSpec{
			Finalizers: internalFinalizers
		},
		Status: projectapi.ProjectStatus{
			Phase: internalNs.Status.Phase,
		},
}
```

No error